packages/mui-styled-engine-sc/src/tools/classnames.js

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable no-continue */
+/* eslint-disable no-plusplus */
+/* eslint-disable import/prefer-default-export */
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+/* eslint-disable no-restricted-syntax */
+
+/** Copy pasted from
+ * https://github.com/emotion-js/emotion/blob/23f43ab9f24d44219b0b007a00f4ac681fe8712e/packages/react/src/class-names.js#L17-L63
+ * */
+export const classnames = (args) => {
+  const len = args.length;
+  let i = 0;
+  let cls = '';
+  for (; i < len; i++) {
+    const arg = args[i];
+    if (arg == null) {
+      continue;
+    }
+
+    let toAdd;
+    switch (typeof arg) {
+      case 'boolean':
+        break;
+      case 'object': {
+        if (Array.isArray(arg)) {
+          toAdd = classnames(arg);
+        } else {
+          if (
+            process.env.NODE_ENV !== 'production' &&
+            arg.styles !== undefined &&
+            arg.name !== undefined
+          ) {
+            console.error(
+              'You have passed styles created with `css` from `@emotion/react` package to the `cx`.\n' +
```

Comment:
Allright.  
Note that this would triggers only if users are using `@emotion/react` despite the fact that they have opted to `@mui/styled-engine-sc`. I think literally no one in this situation so it's dead code.  

That sait what do you feel about making `classnames` a dependency injection?  
...to avoid having duplication of code in `@mui/styled-engine-sc` and `@mui/styled-engine`.  It you don't mind let's leave it at that.

Reply:
Allright.  
Note that this would triggers only if users are using `@emotion/react` despite the fact that they have opted to `@mui/styled-engine-sc`. I think literally no one in this situation so it's dead code.  

That sait what do you feel about making `classnames` a dependency injection?  
...to avoid having duplication of code in `@mui/styled-engine-sc` and `@mui/styled-engine`.  It you don't mind let's leave it at that.

Reply:
I would go with the simplest option possible at the start, just duplicate and remove the warnings. We can optimize later if needed :)

Reply:
Sound wise 👍🏻